src/core/Memory.hpp

```diff
@@ -1,5 +1,9 @@
 #pragma once
 
+extern "C" {
+#include "../platform/platform.h"
+}
+
```

Comment:
It doesn't look like the platform.h file itself is needed, but it's complaining about no declaration of `malloc` and such.